### PR TITLE
feat: only run tests when creating release

### PR DESCRIPTION
**Pull request by Log234**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -8,7 +8,6 @@ pipeline {
   stages {
     stage('Build') {
       environment {
-        GITHUB_TOKEN = credentials('jenkins_release_token')
```
Comment (simendag) [inline Jenkinsfile:11]: Why isn't GITHUB_TOKEN needed anymore?

**Review by simendag — CHANGES_REQUESTED** [review]
See comments.

File: Jenkinsfile (hunk 1 above)
Comment (Log234) [reply Jenkinsfile:11]: Because we don't publish the artifact to GitHub in the build step.

Comment (simendag) [reply Jenkinsfile:11]: Ah, I didn't notice it was there in the publish step. 👍

**Pull request merged**